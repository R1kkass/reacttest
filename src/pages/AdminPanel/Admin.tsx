import { useEffect, useState } from 'react'
import { useSearchParams } from 'react-router-dom'
import AdminPanel from '../../widgets/AdminPanel/AdminPanel'
import EditPanel from '../../widgets/EditFiltres/EditPanel'
import './Admin.scss'

const elArr=[{
    id: 1,
    element: <AdminPanel/>
},
{
    id: 2,
    element: <EditPanel/>
}
]

const Admin = () => {

    const [searchParams, setSearchParams] = useSearchParams()
    const [el, setEl] = useState<any>(null)

    const switchFn = (id: number)=>{
        let res = elArr.filter((el)=>{
            return el.id==id
        })
        setEl(res[0].element)
        searchParams.set('type', String(id))
        setSearchParams(searchParams)
    }

    useEffect(()=>{
        if(searchParams.get('type')){
            switchFn(Number(searchParams.get('type')))
        }
    },[])

    return (
        <div className="Admin" data-testid="admin-page">
            <div className="Admin__tabs">
                <div className={searchParams.get('type')=='1' ? 'Active' : ''} onClick={()=>switchFn(1)}>
                    <h3 >Продукты</h3>
                </div>
                <div className={searchParams.get('type')=='2' ? 'Active' : ''} onClick={()=>switchFn(2)}>
                    <h3>Категории</h3>
                </div>
            </div>
            <div className='Admin__window'>
                {el}
            </div>
        </div>
    )
}

export default Admin
