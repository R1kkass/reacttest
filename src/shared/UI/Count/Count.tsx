import { FC } from "react"
import './Count.scss'
const Count: FC<{
    count: number
    onClickMinus: () => void
    onClickPlus: () => void
}> = ({ onClickMinus, onClickPlus, count }) => {
    return (
        <div className="Count">
            <button onClick={onClickMinus} data-testid="-">-</button>
            <p>{count}</p>
            <button onClick={onClickPlus} data-testid="+">+</button>
        </div>
    )
}

export default Count
