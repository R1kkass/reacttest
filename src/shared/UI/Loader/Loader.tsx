import { createPortal } from "react-dom"
import "./Loader.scss"

const Loader = () => {
    return (
        <>
            {createPortal(
                <div data-testid="loader" className="Loader">
                    <div className="lds-ring">
                        <div></div>
                        <div></div>
                        <div></div>
                        <div></div>
                    </div>
                </div>,
                document.body
            )}
        </>
    )
}

export default Loader
