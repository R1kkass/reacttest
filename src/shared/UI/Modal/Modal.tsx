import { FC, useState } from "react"
import { createPortal } from "react-dom"
import MyButton from "../Buttons/MyButton/MyButton"
import './Modal.scss'

const Modal: FC<{
    children: React.ReactNode
    visible: boolean
    callback?: () => void
}> = ({ visible, children, callback }) => {
    return (
        
        <>
        {visible ? 
        createPortal(
            <div className="Modal" onClick={callback}>
            <div className="Modal__content" onClick={(e:React.MouseEvent)=>e.stopPropagation()}>{children}</div>
        </div>,
            document.body
        )
        :''
        }
        </>
    )
}

export default Modal
