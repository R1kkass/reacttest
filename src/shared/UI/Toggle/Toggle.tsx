import { Children, FC, useState } from "react"
import './Toggle.scss'
import polygonBottom from '../SVG/Polygon/PolygonBottom.svg'
import polygonTop from '../SVG/Polygon/PolygonTop.svg'

const Toggle:FC<{children: React.ReactNode, nameBtn: string, num?: number}> = ({children, nameBtn,num=0}) => {
    const [bol, setBol] = useState<boolean>(false)

    if (bol) {
        return (
            <div className="Toggle">
                <button data-testid="btn" onClick={() => setBol((p) => !p)}>{nameBtn} <img src={polygonTop}/></button>
                <div>
                    {children}
                </div>
            </div>
        )
    }

    return (
        <div className="Toggle">
            <button data-testid="btn" onClick={() => setBol((p) => !p)}>{nameBtn} <img src={polygonBottom}/></button>
        </div>
    )
}

export default Toggle
