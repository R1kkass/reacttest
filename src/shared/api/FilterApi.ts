import axios from "axios"

export interface IFilterApi {
    id?: string
    title: string
    array: IArray[]
    type?: string[]
}

export interface IArray{
    name: string
}

export interface IFilterApiData {
    data: IFilterApi[]
}

export const FilterApi = async () => {
    const res: IFilterApiData = await axios.get(
        "https://641757421cbdda1fa1577617.mockapi.io/cx/xdgf"
    )
    return res
}

export const FilterApiPost = async (post: IFilterApi) => {
    const res: IFilterApiData = await axios.post(
        "https://641757421cbdda1fa1577617.mockapi.io/cx/xdgf",
        { title: post.title, array: post.array }
    )
    return res
}

export const FilterApiDelte = async (id:number) => {
    const res: IFilterApiData = await axios.delete(
        "https://641757421cbdda1fa1577617.mockapi.io/cx/xdgf/"+id
    )
    return res
}

export const FilterApiPut = async (id:number, post: IFilterApi) => {
    const res: IFilterApiData = await axios.put(
        "https://641757421cbdda1fa1577617.mockapi.io/cx/xdgf/"+id,{
            ...post
        }
    )
    return res
}

