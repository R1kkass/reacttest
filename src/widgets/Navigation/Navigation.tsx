import "./Navigation.scss"
import People from "./IMG/pngwing3.png"
import MyButton from "../../shared/UI/Buttons/MyButton/MyButton"
import download from "../../shared/UI/SVG/Download/Download.svg"
import whatsapp from "../../shared/UI/SVG/Messangers/WhatsApp.svg"
import telegram from "../../shared/UI/SVG/Messangers/Telegram.svg"
import visa from "../../shared/UI/SVG/Visa/Visa.svg"
import masterCard from "../../shared/UI/SVG/Visa/MasterCard.svg"
import logo from "../../shared/UI/SVG/Logo/Logo.svg"
import catalog from "../../shared/UI/SVG/Catalog/Catalog.svg"
import address from "../../shared/UI/SVG/Address/Address.svg"
import email from "../../shared/UI/SVG/Email/Email.svg"
import baskets from "../../shared/UI/SVG/Basket/Basket.svg"
import logoFooter from "../../shared/UI/SVG/Logo/LogoFooter.svg"
import { FC, useEffect, useState } from "react"
import MyInput from "../../shared/UI/Input/MyInput"
import { Link } from "react-router-dom"
import { useSelector } from "react-redux"
import { IRedux } from "../../app/Redux/Store/Index"
import { ICardApi } from "../../shared/api/CardApi"
import search from "../../shared/UI/SVG/Search/Search.svg"
import LeftBlockMobile from "../../shared/UI/LeftBlockMobile/LeftBlockMobile"
import mobile from "../../shared/UI/SVG/Mobile/Mobile.svg"

const Navigation: FC<{ children: React.ReactNode }> = ({ children }) => {
    const counts = useSelector((state: IRedux) => state.basket.count)
    const [query, setQuery] = useState<boolean>(false)

    const basket = JSON.parse(localStorage.getItem("basket") || "[]")?.reduce(
        (key: number, count: ICardApi) => {
            console.log(key)
            return (key = key + Number(count.price) * Number(count?.count || 1))
        },
        0
    )

    const bask = useSelector((state: IRedux) => state.basket.basket)

    useEffect(() => {
        var x: any = window.matchMedia("(max-width: 800px)")
        function myFunction(x: any) {
            if (x?.matches) {
                setQuery(true)
            } else {
                setQuery(false)
            }
        }
        myFunction(x)
        x?.addListener(myFunction)

        return x?.removeListener(myFunction)
    }, [])

    const [visible, setVisible] = useState<boolean>(false)

    if (query) {
        return (
            <>
                <LeftBlockMobile
                    visible={visible}
                    setVisible={() => setVisible(false)}
                >
                    <div className="Navigation__address">
                        <img src={address} />
                        <div>
                            <p>г. Кокчетав, ул. Ж. Ташенова 129Б</p>
                            <p>(Рынок Восточный)</p>
                        </div>
                    </div>
                    <div className="Navigation__email">
                        <img src={email} />
                        <div>
                            <p>opt.sultan@mail.ru</p>
                            <p>На связи в любое время</p>
                        </div>
                    </div>
                    <div>
                        <p>О компании</p>
                    </div>
                    <div>
                        <p>Доставка и оплата</p>
                    </div>
                    <div>
                        <Link  data-testid="adm-page"
                         to="/adm">
                            <p>Админ панель</p>
                        </Link>
                    </div>
                    <div>
                        <p>Контакты</p>
                    </div>
                </LeftBlockMobile>
                <div className="Content">
                    <div className="Navigation">
                        <div className="Navigation__firstLine">
                            <div className="Navigation__container"></div>
                            <div
                                onClick={() => setVisible(true)}
                                className="Navigation__ellipse"
                            ><img src={mobile} alt="" /></div>
                            <div>
                                <img src={logo} />
                            </div>
                            <div>
                                <Link to="/basket">
                                    <div className="SecondLine__basket">
                                        <img src={baskets} />
                                        <div className="dottedBasket">
                                            <p>{counts}</p>
                                        </div>
                                    </div>
                                </Link>
                            </div>
                        </div>
                        <div className="Navigation__secondLine">
                            <Link to="/">
                                <div className="SecondLine__button">
                                    <img src={catalog} />
                                    Каталог
                                    <img src={search} />
                                    Поиск
                                </div>
                            </Link>
                        </div>
                    </div>
                    {children}
                </div>
                <div className="Footer">
                    <div className="Footer__logo">
                        <img src={logoFooter} />
                        <p>
                            Компания «Султан» — снабжаем розничные магазины
                            товарами "под ключ" в Кокчетаве и Акмолинской
                            области
                        </p>
                        <p>Подпишись на скидки и акции</p>
                        <div>
                            <MyInput />
                        </div>
                    </div>
                    <div className="Footer__menu">
                        <h2>Меню сайта:</h2>
                        <p>О компании</p>
                        <p>Доставка и оплата</p>
                        <p>Возврат</p>
                        <p>Контакты</p>
                    </div>
                    <div className="Footer__menu">
                        <h2>Категории:</h2>
                        <p>Бытовая химия</p>
                        <p>Косметика и гигиена</p>
                        <p>Товары для дома</p>
                        <p>Товары для детей и мам</p>
                        <p>Посуда</p>
                    </div>
                    <div className="Footer__menu">
                        <h2>Скачать прайс-лист:</h2>
                        <MyButton>
                            Прайс-лист <img src={download}></img>
                        </MyButton>
                        <p>Связь в мессенджерах:</p>
                        <div className="Footer__messanger">
                            <img src={whatsapp}></img>
                            <img src={telegram}></img>
                        </div>
                    </div>
                    <div className="Footer__menu Footer__visa">
                        <h2>Контакты:</h2>
                        <div className="Footer__contactns">
                            <p>+7 (777) 490-00-91</p>
                            <p>время работы: 9:00-20:00</p>
                            <p>Заказать звонок</p>
                        </div>
                        <div className="Footer__contactns2">
                            <p>opt.sultan@mail.ru</p>
                            <p>На связи в любое время</p>
                        </div>

                        <div className="Footer__messanger">
                            <img src={visa}></img>
                            <img src={masterCard}></img>
                        </div>
                    </div>
                </div>
            </>
        )
    }

    return (
        <>
            <div className="Content">
                <div className="Navigation">
                    <div className="Navigation__firstLine">
                        <div className="Navigation__container">
                            <div className="Navigation__address">
                                <img src={address} />
                                <div>
                                    <p>г. Кокчетав, ул. Ж. Ташенова 129Б</p>
                                    <p>(Рынок Восточный)</p>
                                </div>
                            </div>
                            <div className="Navigation__email">
                                <img src={email} />
                                <div>
                                    <p>opt.sultan@mail.ru</p>
                                    <p>На связи в любое время</p>
                                </div>
                            </div>
                        </div>
                        <div className="Navigation__navBar">
                            <div>
                                <p>О компании</p>
                            </div>
                            <div>
                                <p>Доставка и оплата</p>
                            </div>
                            <div>
                                <p>Возврат</p>
                            </div>
                            <div>
                                <p>Контакты</p>
                            </div>
                        </div>
                    </div>
                    <div className="Navigation__secondLine">
                        <img src={logo} />
                        <Link to="/" >
                            <div className="SecondLine__button">
                                <MyButton>
                                    Каталог
                                    <img src={catalog} />
                                </MyButton>
                            </div>
                        </Link>

                        <div className="SecondLine__search">
                            <MyInput />
                        </div>
                        <div className="SecondLine__contact">
                            <div className="SecondLine__number">
                                <p>+7 (777) 490-00-91</p>
                            </div>
                            <div className="SecondLine__time">
                                <p>время работы: 9:00-20:00</p>
                            </div>
                            <div className="SecondLine__call">
                                <p>Заказать звонок</p>
                            </div>
                            <div className="SecondLine__img">
                                <img src={People} />
                                <div className="dottedGreen"></div>
                            </div>
                        </div>
                        <div>
                            <MyButton>
                                Прайс-лист
                                <img src={download} />
                            </MyButton>
                        </div>
                        <Link to="/basket" data-testid="main-link">
                            <div className="SecondLine__basket">
                                <img src={baskets} />
                                <div className="dottedBasket">
                                    <p>{counts}</p>
                                </div>
                            </div>
                        </Link>
                        <Link to="/basket">
                            <div className="SecondLine__price">
                                <p>Корзина</p>
                                <p>
                                    {bask?.reduce((key, count) => {
                                        return (key +=
                                            Number(count.price) *
                                            (count?.count || 1))
                                    }, 0) || basket}
                                    ₸
                                </p>
                            </div>
                        </Link>
                    </div>
                </div>
                {children}
            </div>
            <div className="Footer">
                <div className="Footer__logo">
                    <img src={logoFooter} />
                    <p>
                        Компания «Султан» — снабжаем розничные магазины товарами
                        "под ключ" в Кокчетаве и Акмолинской области
                    </p>
                    <p>Подпишись на скидки и акции</p>
                    <div>
                        <MyInput />
                    </div>
                </div>
                <div className="Footer__menu">
                    <h2>Меню сайта:</h2>
                    <p>О компании</p>
                    <p>Доставка и оплата</p>
                    <p>Возврат</p>
                    <p>Контакты</p>
                </div>
                <div className="Footer__menu">
                    <h2>Категории:</h2>
                    <p>Бытовая химия</p>
                    <p>Косметика и гигиена</p>
                    <p>Товары для дома</p>
                    <p>Товары для детей и мам</p>
                    <p>Посуда</p>
                </div>
                <div className="Footer__menu">
                    <h2>Скачать прайс-лист:</h2>
                    <MyButton>
                        Прайс-лист <img src={download}></img>
                    </MyButton>
                    <p>Связь в мессенджерах:</p>
                    <div className="Footer__messanger">
                        <img src={whatsapp}></img>
                        <img src={telegram}></img>
                    </div>
                </div>
                <div className="Footer__menu Footer__visa">
                    <h2>Контакты:</h2>
                    <div className="Footer__contactns">
                        <p>+7 (777) 490-00-91</p>
                        <p>время работы: 9:00-20:00</p>
                        <p>Заказать звонок</p>
                    </div>
                    <div className="Footer__contactns2">
                        <p>opt.sultan@mail.ru</p>
                        <p>На связи в любое время</p>
                    </div>

                    <div className="Footer__messanger">
                        <img src={visa}></img>
                        <img src={masterCard}></img>
                    </div>
                </div>
            </div>
        </>
    )
}

export default Navigation
