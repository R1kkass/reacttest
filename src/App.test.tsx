import React, { useState } from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import App from './App';
import Basket from './pages/Basket/Basket';
import { Provider, useSelector } from 'react-redux';
import {store} from '../src/app/Redux/Store/Index'
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { addBasket } from './app/Redux/Store/basket';
import { CardApi, ICardApi, CardApiId, ICardData, DeleteCardApi, AddCardApi } from './shared/api/CardApi';
import Navigation from './widgets/Navigation/Navigation';
import userEvent from '@testing-library/user-event';
import Admin from './pages/AdminPanel/Admin';
import Routing from './pages';
import MyButton from './shared/UI/Buttons/MyButton/MyButton';
import Count from './shared/UI/Count/Count';
import Modal from './shared/UI/Modal/Modal';
import Loader from './shared/UI/Loader/Loader';
import CheckBox from './shared/UI/CheckBox/CheckBox';
import Toggle from './shared/UI/Toggle/Toggle';

const res:ICardApi[] = [{
  id: 1,
  imgURL: '',
  size: '',
  brand: '',
  code: '',
  manufacturer: '43',
  price: 432,
  name: '',
  count:1,
  description:'',
  type: ['fd','fd']
}]


describe("Test", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      }))
    });
  });



  test('Empty arr', () => {
    render(<MemoryRouter><Provider store={store}><Basket /></Provider></MemoryRouter>);
    let res = screen.getByText(/Корзина пуста/i)
    expect(res).toBeInTheDocument();
  });

  test('MyButton', async() => {
    render(
    <MyButton >Кнопка</MyButton>
    );
    let link = screen.getByText(/Кнопка/i)
    expect(link).toBeInTheDocument();
  });

  test('Modal', ()=>{

    render(<Modal visible={true}>
      <div>Модалка</div>
    </Modal>)
    let link = screen.getByText(/Модалка/i)
    expect(link).toBeInTheDocument();
  })

  test('Loader', async ()=>{
    render(<Loader/>)
    let link = await screen.findAllByTestId("loader")
    expect(link[0]).toBeInTheDocument();

  })

  test('CardApi', async ()=>{
    let res = await CardApi()
    expect(res.data).toHaveLength(19)
  })

  test('AddCardApi', async ()=>{
    let res = await AddCardApi({
      id: 31,
      imgURL: 'URLid',
      size: '321ml',
      brand: 'ds',
      code: '3213',
      manufacturer: 'dsasda',
      price: 321,
      name: 'dsa',
      count: 1,
      description: 'dsadsa',
      type: ['dsa']
    })
    expect(res.data?.imgURL).toEqual('URLid')
  })

  test('DeleteCardApi', async ()=>{
    let res = await DeleteCardApi(31)
    expect(res.data?.imgURL).toEqual('URLid')
  })

  test('CardApiId', async()=>{
    let res = await CardApiId('6')
    expect(res.data?.code).toEqual('4532543253')
  })

  test('CheckBox', ()=>{
    render(<CheckBox text="text1" count="1"/>)
    expect(screen.getByText(/text1/i)).toBeInTheDocument();
  })

  test('Toggle', async ()=>{
    render(
      <Toggle nameBtn='Кнопка'>
        <div>Tog</div>
      </Toggle>
    )
    let btn = screen.getByTestId('btn')
    fireEvent.click(btn)
    expect(screen.getByText(/Tog/i)).toBeInTheDocument();
  })
});