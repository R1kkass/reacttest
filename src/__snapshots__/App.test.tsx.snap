// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Navigation 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Content"
      >
        <div
          class="Navigation"
        >
          <div
            class="Navigation__firstLine"
          >
            <div
              class="Navigation__container"
            >
              <div
                class="Navigation__address"
              >
                <img
                  src="Address.svg"
                />
                <div>
                  <p>
                    г. Кокчетав, ул. Ж. Ташенова 129Б
                  </p>
                  <p>
                    (Рынок Восточный)
                  </p>
                </div>
              </div>
              <div
                class="Navigation__email"
              >
                <img
                  src="Email.svg"
                />
                <div>
                  <p>
                    opt.sultan@mail.ru
                  </p>
                  <p>
                    На связи в любое время
                  </p>
                </div>
              </div>
            </div>
            <div
              class="Navigation__navBar"
            >
              <div>
                <p>
                  О компании
                </p>
              </div>
              <div>
                <p>
                  Доставка и оплата
                </p>
              </div>
              <div>
                <p>
                  Возврат
                </p>
              </div>
              <div>
                <p>
                  Контакты
                </p>
              </div>
            </div>
          </div>
          <div
            class="Navigation__secondLine"
          >
            <img
              src="Logo.svg"
            />
            <a
              href="/"
            >
              <div
                class="SecondLine__button"
              >
                <button
                  class="MyButton"
                >
                  Каталог
                  <img
                    src="Catalog.svg"
                  />
                </button>
              </div>
            </a>
            <div
              class="SecondLine__search"
            >
              <div
                class="MyInput"
              >
                <input
                  placeholder="Поиск..."
                />
                <button
                  class="MyButton"
                >
                  <img
                    src="Search.svg"
                  />
                </button>
              </div>
            </div>
            <div
              class="SecondLine__contact"
            >
              <div
                class="SecondLine__number"
              >
                <p>
                  +7 (777) 490-00-91
                </p>
              </div>
              <div
                class="SecondLine__time"
              >
                <p>
                  время работы: 9:00-20:00
                </p>
              </div>
              <div
                class="SecondLine__call"
              >
                <p>
                  Заказать звонок
                </p>
              </div>
              <div
                class="SecondLine__img"
              >
                <img
                  src="pngwing3.png"
                />
                <div
                  class="dottedGreen"
                />
              </div>
            </div>
            <div>
              <button
                class="MyButton"
              >
                Прайс-лист
                <img
                  src="Download.svg"
                />
              </button>
            </div>
            <a
              href="/basket"
            >
              <div
                class="SecondLine__basket"
              >
                <img
                  src="Basket.svg"
                />
                <div
                  class="dottedBasket"
                >
                  <p>
                    0
                  </p>
                </div>
              </div>
            </a>
            <a
              href="/basket"
            >
              <div
                class="SecondLine__price"
              >
                <p>
                  Корзина
                </p>
                <p>
                  0
                  ₸
                </p>
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        class="Footer"
      >
        <div
          class="Footer__logo"
        >
          <img
            src="LogoFooter.svg"
          />
          <p>
            Компания «Султан» — снабжаем розничные магазины товарами "под ключ" в Кокчетаве и Акмолинской области
          </p>
          <p>
            Подпишись на скидки и акции
          </p>
          <div>
            <div
              class="MyInput"
            >
              <input
                placeholder="Поиск..."
              />
              <button
                class="MyButton"
              >
                <img
                  src="Search.svg"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="Footer__menu"
        >
          <h2>
            Меню сайта:
          </h2>
          <p>
            О компании
          </p>
          <p>
            Доставка и оплата
          </p>
          <p>
            Возврат
          </p>
          <p>
            Контакты
          </p>
        </div>
        <div
          class="Footer__menu"
        >
          <h2>
            Категории:
          </h2>
          <p>
            Бытовая химия
          </p>
          <p>
            Косметика и гигиена
          </p>
          <p>
            Товары для дома
          </p>
          <p>
            Товары для детей и мам
          </p>
          <p>
            Посуда
          </p>
        </div>
        <div
          class="Footer__menu"
        >
          <h2>
            Скачать прайс-лист:
          </h2>
          <button
            class="MyButton"
          >
            Прайс-лист 
            <img
              src="Download.svg"
            />
          </button>
          <p>
            Связь в мессенджерах:
          </p>
          <div
            class="Footer__messanger"
          >
            <img
              src="WhatsApp.svg"
            />
            <img
              src="Telegram.svg"
            />
          </div>
        </div>
        <div
          class="Footer__menu Footer__visa"
        >
          <h2>
            Контакты:
          </h2>
          <div
            class="Footer__contactns"
          >
            <p>
              +7 (777) 490-00-91
            </p>
            <p>
              время работы: 9:00-20:00
            </p>
            <p>
              Заказать звонок
            </p>
          </div>
          <div
            class="Footer__contactns2"
          >
            <p>
              opt.sultan@mail.ru
            </p>
            <p>
              На связи в любое время
            </p>
          </div>
          <div
            class="Footer__messanger"
          >
            <img
              src="Visa.svg"
            />
            <img
              src="MasterCard.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Content"
    >
      <div
        class="Navigation"
      >
        <div
          class="Navigation__firstLine"
        >
          <div
            class="Navigation__container"
          >
            <div
              class="Navigation__address"
            >
              <img
                src="Address.svg"
              />
              <div>
                <p>
                  г. Кокчетав, ул. Ж. Ташенова 129Б
                </p>
                <p>
                  (Рынок Восточный)
                </p>
              </div>
            </div>
            <div
              class="Navigation__email"
            >
              <img
                src="Email.svg"
              />
              <div>
                <p>
                  opt.sultan@mail.ru
                </p>
                <p>
                  На связи в любое время
                </p>
              </div>
            </div>
          </div>
          <div
            class="Navigation__navBar"
          >
            <div>
              <p>
                О компании
              </p>
            </div>
            <div>
              <p>
                Доставка и оплата
              </p>
            </div>
            <div>
              <p>
                Возврат
              </p>
            </div>
            <div>
              <p>
                Контакты
              </p>
            </div>
          </div>
        </div>
        <div
          class="Navigation__secondLine"
        >
          <img
            src="Logo.svg"
          />
          <a
            href="/"
          >
            <div
              class="SecondLine__button"
            >
              <button
                class="MyButton"
              >
                Каталог
                <img
                  src="Catalog.svg"
                />
              </button>
            </div>
          </a>
          <div
            class="SecondLine__search"
          >
            <div
              class="MyInput"
            >
              <input
                placeholder="Поиск..."
              />
              <button
                class="MyButton"
              >
                <img
                  src="Search.svg"
                />
              </button>
            </div>
          </div>
          <div
            class="SecondLine__contact"
          >
            <div
              class="SecondLine__number"
            >
              <p>
                +7 (777) 490-00-91
              </p>
            </div>
            <div
              class="SecondLine__time"
            >
              <p>
                время работы: 9:00-20:00
              </p>
            </div>
            <div
              class="SecondLine__call"
            >
              <p>
                Заказать звонок
              </p>
            </div>
            <div
              class="SecondLine__img"
            >
              <img
                src="pngwing3.png"
              />
              <div
                class="dottedGreen"
              />
            </div>
          </div>
          <div>
            <button
              class="MyButton"
            >
              Прайс-лист
              <img
                src="Download.svg"
              />
            </button>
          </div>
          <a
            href="/basket"
          >
            <div
              class="SecondLine__basket"
            >
              <img
                src="Basket.svg"
              />
              <div
                class="dottedBasket"
              >
                <p>
                  0
                </p>
              </div>
            </div>
          </a>
          <a
            href="/basket"
          >
            <div
              class="SecondLine__price"
            >
              <p>
                Корзина
              </p>
              <p>
                0
                ₸
              </p>
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      class="Footer"
    >
      <div
        class="Footer__logo"
      >
        <img
          src="LogoFooter.svg"
        />
        <p>
          Компания «Султан» — снабжаем розничные магазины товарами "под ключ" в Кокчетаве и Акмолинской области
        </p>
        <p>
          Подпишись на скидки и акции
        </p>
        <div>
          <div
            class="MyInput"
          >
            <input
              placeholder="Поиск..."
            />
            <button
              class="MyButton"
            >
              <img
                src="Search.svg"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="Footer__menu"
      >
        <h2>
          Меню сайта:
        </h2>
        <p>
          О компании
        </p>
        <p>
          Доставка и оплата
        </p>
        <p>
          Возврат
        </p>
        <p>
          Контакты
        </p>
      </div>
      <div
        class="Footer__menu"
      >
        <h2>
          Категории:
        </h2>
        <p>
          Бытовая химия
        </p>
        <p>
          Косметика и гигиена
        </p>
        <p>
          Товары для дома
        </p>
        <p>
          Товары для детей и мам
        </p>
        <p>
          Посуда
        </p>
      </div>
      <div
        class="Footer__menu"
      >
        <h2>
          Скачать прайс-лист:
        </h2>
        <button
          class="MyButton"
        >
          Прайс-лист 
          <img
            src="Download.svg"
          />
        </button>
        <p>
          Связь в мессенджерах:
        </p>
        <div
          class="Footer__messanger"
        >
          <img
            src="WhatsApp.svg"
          />
          <img
            src="Telegram.svg"
          />
        </div>
      </div>
      <div
        class="Footer__menu Footer__visa"
      >
        <h2>
          Контакты:
        </h2>
        <div
          class="Footer__contactns"
        >
          <p>
            +7 (777) 490-00-91
          </p>
          <p>
            время работы: 9:00-20:00
          </p>
          <p>
            Заказать звонок
          </p>
        </div>
        <div
          class="Footer__contactns2"
        >
          <p>
            opt.sultan@mail.ru
          </p>
          <p>
            На связи в любое время
          </p>
        </div>
        <div
          class="Footer__messanger"
        >
          <img
            src="Visa.svg"
          />
          <img
            src="MasterCard.svg"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`With arr 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Basket"
      >
        <div
          class="BreadCrumbs"
        >
          <a
            href="/"
          >
            Каталог
             
            <img
              src="Crumb.svg"
            />
             
          </a>
          <a
            href="/basket"
          >
            Корзина
             
            <img
              src="Crumb.svg"
            />
             
          </a>
        </div>
        <h1>
          Корзина
        </h1>
        <div
          class="Basket__order"
        >
          <h1>
            Корзина пуста
          </h1>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Basket"
    >
      <div
        class="BreadCrumbs"
      >
        <a
          href="/"
        >
          Каталог
           
          <img
            src="Crumb.svg"
          />
           
        </a>
        <a
          href="/basket"
        >
          Корзина
           
          <img
            src="Crumb.svg"
          />
           
        </a>
      </div>
      <h1>
        Корзина
      </h1>
      <div
        class="Basket__order"
      >
        <h1>
          Корзина пуста
        </h1>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
